server/service/src/asset/query_log.rs

```diff
@@ -0,0 +1,38 @@
+use repository::assets::asset_log::{AssetLog, AssetLogFilter, AssetLogRepository, AssetLogSort};
+use repository::{EqualFilter, PaginationOption, StorageConnection};
+
+use crate::{
+    get_default_pagination, i64_to_u32, service_provider::ServiceContext, ListError, ListResult,
+    SingleRecordError,
+};
+
+pub const MAX_LIMIT: u32 = 1000;
+pub const MIN_LIMIT: u32 = 1;
+
+pub fn get_asset_logs(
+    connection: &StorageConnection,
+    pagination: Option<PaginationOption>,
+    filter: Option<AssetLogFilter>,
+    sort: Option<AssetLogSort>,
+) -> Result<ListResult<AssetLog>, ListError> {
+    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
+    let repository = AssetLogRepository::new(&connection);
+
+    Ok(ListResult {
+        rows: repository.query(pagination, filter.clone(), sort)?,
+        count: i64_to_u32(repository.count(filter)?),
+    })
+}
+
+pub fn get_asset_log(ctx: &ServiceContext, id: String) -> Result<AssetLog, SingleRecordError> {
+    let repository = AssetLogRepository::new(&ctx.connection);
+
+    let mut result =
+        repository.query_by_filter(AssetLogFilter::new().id(EqualFilter::equal_to(&id)))?;
+
+    if let Some(record) = result.pop() {
+        Ok(record)
+    } else {
+        Err(SingleRecordError::NotFound(id))
+    }
```

Comment:
I am wondering now here about the absence of permissions to access these logs. Should they be restricted? I'm wondering if it might be something different to server admin.

Reply:
Yes it will be, we need to discuss how we'll setup the permissions next week, please remind us!